src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerActionCounts.cs

```diff
@@ -174,6 +179,11 @@ internal class AnalyzerActionCounts
         /// </summary>
         public bool HasAnyExecutableCodeActions { get; }
 
+        /// <summary>
+        /// Returns true if there are any actions that require any semantic or symbolic analysis.
+        /// </summary>
+        public bool HasAnyActionsRequiringSemanticAnalysis { get; }
```

Comment:
I'm quite confused because `SyntaxNodeAnalysisContext` has `SemanticModel` property and can require semantic analysis (depending on how the analyzer is written). How things work in this case? Is `CompilationEventQueue` doesn't have to be populated for this case?

Reply:
> How things work in this case? Is CompilationEventQueue doesn't have to be populated for this case?

It does have to be populated in this case, i.e. non-zero SyntaxNodeActionsCount. Added code does account for it.

Reply:
> It does have to be populated in this case, i.e. non-zero SyntaxNodeActionsCount. Added code does account for it.

@mavasani Is the code not pushed yet?
Also what about compilation start, which has access to `Compilation` and almost always ask for symbols? Is it okay to skip `GetDiagnostics` for this?

Reply:
@Youssef1313 the property here is determining if compilation events are needed to be generated to ensure analyzer driver makes the appropriate callbacks. SyntaxTreeActions, CompilationStartedAction, AdditionalFileActions, are all always executed by the driver and don’t need any events to enqueued into the compilation event queue. Maybe I should adjust the name of the new property being added here and also add more descriptive doc comment to make this more clear.